packages/runtime/src/useCases/transport/messages/SendMessage.ts

```diff
@@ -73,21 +74,81 @@ export class SendMessageUseCase extends UseCase<SendMessageRequest, MessageDTO>
+                    );
+                }
+                return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                    `An active Relationship with the given address '${missingOrInactiveRelationshipAddressArray[0]}' does not exist.`
+                );
+            }
+            if (missingOrInactiveRelationshipAddressArray[1]) {
+                if (peerInDeletionAddressArray.length > 0 && !peerInDeletionAddressArray[1]) {
+                    return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                        `An active Relationship with the given addresses '${missingOrInactiveRelationshipAddressArray}' do not exist and the recipient with the address '${peerInDeletionAddressArray[0]}' has an active IdentityDeletionProcess, so you cannot send them a Message.`
+                    );
+                }
+                if (peerInDeletionAddressArray[1]) {
+                    return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                        `An active Relationship with the given addresses '${missingOrInactiveRelationshipAddressArray}' do not exist and the recipients with the addresses '${peerInDeletionAddressArray}' have an active IdentityDeletionProcess, so you cannot send them a Message.`
+                    );
+                }
+                return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                    `An active Relationship with the given addresses '${missingOrInactiveRelationshipAddressArray}' do not exist.`
+                );
+            }
         }
 
-        if (recipients.length > 1) return RuntimeErrors.general.invalidPropertyValue("Only one recipient is allowed for sending Requests.");
+        if (peerInDeletionAddressArray.length > 0 && !peerInDeletionAddressArray[1]) {
+            return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                `The recipient with the address '${peerInDeletionAddressArray[0]}' has an active IdentityDeletionProcess, so you cannot send them a Message.`
+            );
+        }
 
-        const recipient = CoreAddress.from(recipients[0]);
-        if (!recipient.equals(localRequest.peer)) return RuntimeErrors.general.invalidPropertyValue("The recipient does not match the Request's peer.");
+        if (peerInDeletionAddressArray.length > 0 && peerInDeletionAddressArray[1]) {
+            return TransportCoreErrors.messages.missingOrWrongRelationshipStatusOrPeerInDeletion(
+                `The recipients with the following addresses '${peerInDeletionAddressArray}' have an active IdentityDeletionProcess, so you cannot send them a Message.`
+            );
```

Comment:
This code has to be refactored massively, it is way to complex. 
I think you should start by extracting this to a method and then trying to apply the bouncer / early-return pattern.